content/02_KeyTypes.md

```diff
@@ -1,488 +1,531 @@
+
+Observant readers will have noticed that an example in the preceding chapter looks like it shouldn't work. Having created an `IObservable<long>` that produced events once per second, it subscribed to it with this code:
+
+```cs
+ticks.Subscribe(
+    tick => Console.WriteLine($"Tick {tick}"));
+```
+
+That's passing a delegate, but we can see that `IObservable<T>.Subscribe` requires something called an `IObserver<T>`. We'll get to `IObserver<T>` shortly, but all that's happening here is that this example is using an extension method from the Rx NuGet package:
+
+```cs
+// From the System.Reactive library's ObservableExtensions class
+public static IDisposable Subscribe<T>(this IObservable<T> source, Action<T> onNext)
+```
+
+This is a helper method that wraps a delegate in an implementation of `IObserver<T>` and then passes that to `IObservable<T>.Subscribe`. The effect is that we can write just a simple method (instead of a complete implementation of `IObserver<T>`) and the observable source will invoke our callback each time it wants to supply a value.
+
+Since `IObservable<T>` defines just one method, `Subscribe`, you might be wondering how we can do anything interesting with it. The power comes from the LINQ operators that the `System.Reactive` NuGet library supplies.
+
+### LINQ Operators and Composition
+
+So far I've shown only a very simple LINQ example, using the `Where` operator to filter events down to ones that meet certain criteria. To give you a flavour of how we can build more advanced functionality through composition, I'm going to introduce an example scenario.
+
+Suppose you want to write a program that watches some folder on a filesystem, and performs automatic processing any time something in that folder changes. For example, web developers often to trigger automatic rebuilds of their client side code when they save changes in the editor so they can quickly see the effect of their changes. Often in these cases, filesystem changes often come in bursts—text editors might perform a few distinct operations when saving a file. (Some save modifications to a new file, then perform a couple of renames once this is complete, because this avoids data loss if a power failure or system crash happens to occur at the moment you save the file.) So you typically won't want to take action as soon as you detect file activity. It would be better to give it a moment to see if any more activity occurs, and take action only after everything has settled down.
+
+So what we really want to report is those moments when everything goes quiet after a flurry of activity. The following code defines an Rx operator that detects and reports such things. If you're new to Rx (which seems likely if you're reading this) it probably won't be instantly obvious how this works, so I'll walk through it.
```

Comment:
is it worth specifying it's a custom Rx operator - otherwise people might assume it's an native one. Actually at this point have you actually described what an operator is?